WordPress/gutenberg
# Serializer: Remove HTML beautification, preserve whitespace

Supersedes #6716 
Fixes #4456

This pull request seeks to remove in-block HTML beautification. Per intent with #633, there are still newlines between blocks. Within a block, however, the default serialization will not inject any of its own automated whitespace or tabbing. Existing intentional whitespace will be preserved (e.g. code blocks, shortcode blocks).

This has a few benefits:

- Performance
   - `js-beautify` accounts for 48kb (18.1kb gzipped), now removed
   - No longer need to parse / rewrite HTML when saving a post during serialization stage
- Accuracy
   - `children` parse would include (often wrongly) text nodes not always intended but automatically injected by beautifier
   - No more collapsing whitespace in shortcode blocks
      - Verified in updated `adding-blocks` E2E test which would have previously failed in master (see bd78d55)
   - Surfaces potential issues with raw handler not stripping newlines where stripping may be intended (see new `test/integration/fixtures` output files, e.g. `apple-out.html` table block)

**Implementation notes:**

The vast majority of affected file changes are fixture updates. The only meaningful changes are those made within `blocks/api/serializer.js`, though fixtures verification would be good as well.

**Testing instructions:**

Verify there are no issues with serializing and restoring block contents.

Examples:

 - Save a Demo post and reload it, ensure no validation errors
 - Save a post including a Columns block and reload it, ensure content preserved and no validation errors
